Limber/MIXMockServer.cpp

```diff
@@ -0,0 +1,277 @@
+            serverFromClientConnectionServer_ = BoostConnectionServer::create(5222, boostIOServiceThread_.getIOService(), eventLoop);
+            serverFromClientConnectionServer_->onNewConnection.connect(boost::bind(&Server::handleNewConnection, this, _1));
+            serverFromClientConnectionServer_->start();
+        }
+
+    private:
+        void handleNewConnection(std::shared_ptr<Connection> c) {
+            auto session  = std::make_shared<ServerFromClientSession>(idGenerator_.generateID(), c, &payloadParserFactories_, &payloadSerializers_, &xmlParserFactory, userRegistry_);
+            serverFromClientSessions_.push_back(session);
+            session->onElementReceived.connect(boost::bind(&Server::handleElementReceived, this, _1, session));
+            session->onSessionFinished.connect(boost::bind(&Server::handleSessionFinished, this, session));
+            session->startSession();
+        }
+
+        void handleSessionFinished(std::shared_ptr<ServerFromClientSession> session) {
+            serverFromClientSessions_.erase(std::remove(serverFromClientSessions_.begin(), serverFromClientSessions_.end(), session), serverFromClientSessions_.end());
+        }
+
+        void handleElementReceived(std::shared_ptr<ToplevelElement> element, std::shared_ptr<ServerFromClientSession> session) {
+            std::shared_ptr<Stanza> stanza(std::dynamic_pointer_cast<Stanza>(element));
+            if (!stanza) {
+                return;
+            }
+            stanza->setFrom(session->getRemoteJID());
+
+            auto iq = std::dynamic_pointer_cast<IQ>(stanza);
+            if (!iq) {
+                return;
+            }
+
+            if(iq->getPayload<RosterPayload>()) {
+                SWIFT_LOG(debug) << "Query: Roster" << std::endl;
+                auto i = rosterMap_.find(iq->getFrom().toBare());
+                if (i != rosterMap_.end()) {
+                    SWIFT_LOG(debug) << "Query: Roster Found." << std::endl;
+                    session->sendElement(IQ::createResult(iq->getFrom(), iq->getID(), getRosterPayloadFromXMPPRoster(i->second)));
+                } else {
+                    SWIFT_LOG(debug) << "Query: Roster Not Found. Creating New." << std::endl;
+                    auto newRoster = std::make_shared<XMPPRosterImpl>();
+                    rosterMap_.insert(std::make_pair(iq->getFrom().toBare(), newRoster));
+                    session->sendElement(IQ::createResult(iq->getFrom(), iq->getID(), getRosterPayloadFromXMPPRoster(newRoster)));
+                }
+                return;
+            }
+
+            //If request comes to particular channel supported by service.
+            if (mixChannelRegistry_->hasMIXChannel(stanza->getTo())) {
+                handleElementReceivedForMIXChannel(iq, session);
+            }
+
+            //If request comes to domain service
```

Comment:
Could you move the comments for these if inside the if block so that the `else if (…` directly follows the line with the `}`, thanks. The current style makes it appear an if statement has ended while the corresponding else is following up a couple lines below.